src/client.cpp

```diff
@@ -1770,25 +1764,17 @@ void Client::makeScreenshot()
 	if (serial == SCREENSHOT_MAX_SERIAL_TRIES) {
 		infostream << "Could not find suitable filename for screenshot" << std::endl;
 	} else {
-		irr::video::IImage* const image =
-				driver->createImage(video::ECF_R8G8B8, raw_image->getDimension());
-
-		if (image) {
-			raw_image->copyTo(image);
+		std::string message;
+		if (RenderingEngine::take_screenshot(filename))
+			message = strgettext("Saved screenshot to '%s'");
+		else
+			message = strgettext("Failed to take/save screenshot '%s'");
 
-			std::ostringstream sstr;
-			if (driver->writeImageToFile(image, filename.c_str(), quality)) {
-				sstr << "Saved screenshot to '" << filename << "'";
-			} else {
-				sstr << "Failed to save screenshot '" << filename << "'";
-			}
-			pushToChatQueue(narrow_to_wide(sstr.str()));
-			infostream << sstr.str() << std::endl;
-			image->drop();
-		}
+		char message_buf[200];
+		snprintf(message_buf, sizeof(message_buf), message.c_str(), filename.c_str());
+		pushToChatQueue(narrow_to_wide(std::string(message_buf)));
```

Comment:
why not use wgettext if you want widestring ? it sounds better (don't forget to remove the pointed data)